randovania/generator/teleporter_distributor.py

```diff
@@ -76,18 +76,17 @@ def try_randomize_teleporters(
         teleporter_list.remove(source_teleporter)
         teleporter_list.remove(target_teleporter)
 
-    # TODO
-    list3 = copy.copy(telepoter_database)
+    list3 = copy.copy(teleporter_database)
     cteleporter_list3 = [list3[0]]
     while list3:
         cteleporter_list1 = []
         for cteleporter1 in cteleporter_list3:
             index = 0
             while index < len(list3):
                 cteleporter2 = list3[index]
-                if (
-                    cteleporter2.region_name == cteleporter1.region_name
-                    or cteleporter2.area_name == cteleporter1.destination.area
+                if cteleporter2.region_name == cteleporter1.region_name or (
+                    cteleporter2.region_name == cteleporter1.destination.region
+                    and cteleporter2.area_name == cteleporter1.destination.area
```

Comment:
Would be nice to comment this, so that the next person looking into this doesnt have to find this PR to figure out what this is supposed to achieve.

Reply:
Added a few comments + gave variables a proper name instead of `list1,2,3` and `cteleporter1,2,3` etc.